pydantic/_internal/_generate_schema.py

```diff
@@ -235,6 +238,13 @@ def _add_custom_serialization_from_json_encoders(
         if encoder is None:
             continue
 
+        from pydantic.version import VERSION
+
+        warnings.warn(
+            f'`json_encoders` is deprecated. See https://docs.pydantic.dev/{VERSION}/usage/serialization/#custom-serializers for alternatives',
+            PydanticDeprecatedSince20,
```

Comment:
this is incorrect, it should only be the major and minor versions.

Maybe the solution is to have a proper warnings page like we have for user and validation errors. I can work on this today.

Fixing this should not block release.

Reply:
It actually redirects, e.g. https://docs.pydantic.dev/2.0.2/usage/serialization/#custom-serializers works

Reply:
only for specific versions see #6527.

I'll provide a general fix.

Reply:
Ah I see. Okay then we can just strip the patch version.